src/OpenTelemetry.Api/Context/Propagation/TextMapPropagator.cs

```diff
@@ -42,6 +42,18 @@ public abstract class TextMapPropagator
         /// <param name="setter">Action that will set name and value pair on the object.</param>
         public abstract void Inject<T>(PropagationContext context, T carrier, Action<T, string, string> setter);
 
+        /// <summary>
+        /// Injects <see cref="PropagationContext"/> into a carrier.
+        /// </summary>
+        /// <typeparam name="T">Type of an object to set context on. Typically HttpRequest or similar.</typeparam>
+        /// <param name="context">The default context to transmit over the wire.</param>
+        /// <param name="carrier">Object to set context on. Instance of this object will be passed to setter.</param>
+        /// <param name="setter">Action that will set name and value pair on the object.</param>
+        public virtual void Inject<T>(in PropagationContext context, T carrier, Action<T, string, string> setter)
```

Comment:
The base Inject & Extract are virtual with implementations that call the existing abstract methods which make copies. This is the backwards compatibility mechanism. Propagators will need to opt-into the new path by overriding these default implementations.